### PR TITLE
add command-line flags for agent command(#152)

### LINKED ISSUE #152: Add flags that can set all configurations of Agent
**What would you like to be added**:

Currently, users can only set the configurations by passing the [config file flag](https://yorkie.dev/docs/main/commands)(`-c`). We need to add flags to [agent cmd](https://github.com/yorkie-team/yorkie/blob/main/pkg/cli/agent.go#L116) so that we can set all the configurations of the agent individually.

**Why is this needed**:

It would be convenient if users could directly set the agent's configuration as flags in a deployment environment such as Docker Compose or Kubernetes.

### PR DESCRIPTION
**Pull request by computerphilosopher**
**What this PR does / why we need it**:

This PR adds long flags that can set all configurations of Agent. 

**Which issue(s) this PR fixes**:

Fixes #152 

**Special notes for your reviewer**:

It assigns command-line flags directly to the configuration struct variable. 

timeout-second variables don't be assigned directly from command-line flags. Because to assign duration variable from command-line arguments, users have to specify the time unit. (5s, 1m...) this form doesn't fit with the flag's name(timeout-'second'.)

ETCD endpoints don't be assigned directly too. Because etcd config's pointer is nil for default. 

**Does this PR introduce a user-facing change?**:

```release-note
Add flags that can set all configurations of Agent
```

**Additional documentation**:

```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: pkg/cli/agent.go [hunk 1]
```diff
@@ -120,5 +132,73 @@ func init() {
...
+	cmd.Flags().StringVar(
+		&conf.Mongo.ConnectionURI,
+		"mongo-connection-uri",
+		yorkie.DefaultMongoConnectionURI,
+		"MongoDB's connection URI",
+	)
+	cmd.Flags().StringVar(
+		&conf.Mongo.YorkieDatabase,
+		"mongo-yorkie-database",
+		yorkie.DefaultMongoConnectionURI,
```
Comment (hackerwins) [inline pkg/cli/agent.go:174]:
```suggestion
		yorkie.DefaultMongoYorkieDatabase,
```

File: pkg/cli/agent.go [hunk 2]
```diff
@@ -27,22 +27,34 @@ import (
...
 		RunE: func(cmd *cobra.Command, args []string) error {
-			conf := yorkie.NewConfig()
+			conf.Mongo.ConnectionTimeoutSec = time.Duration(mongoConnectionTimeoutSec)
+			conf.Mongo.PingTimeoutSec = time.Duration(mongoPingTimeoutSec)
+			if etcdEndpoints != nil {
+				conf.ETCD = &etcd.Config{
+					Endpoints: etcdEndpoints,
+				}
+			}
+			//if config file is given, command-line argumentes will be overwritten
```
Comment (hackerwins) [inline pkg/cli/agent.go:57]:
```suggestion
			// If config file is given, command-line arguments will be overwritten.
```

File: pkg/cli/agent.go [hunk 3]
```diff
@@ -120,5 +132,73 @@ func init() {
 		"",
 		"config path",
```
Comment (hackerwins) [inline pkg/cli/agent.go:133]:
```suggestion
		"Config path",
```

File: pkg/cli/agent.go [hunk 4]
```diff
@@ -120,5 +132,73 @@ func init() {
...
+		&conf.RPC.CertFile,
+		"rpc-key-file",
+		"",
+		"RPC key file's path",
+	)
+	cmd.Flags().IntVar(
+		&conf.Metrics.Port,
+		"metrics-port",
+		yorkie.DefaultMetricsPort,
+		"metrics port",
```
Comment (hackerwins) [inline pkg/cli/agent.go:157]:
```suggestion
		"Metrics port",
```

**Review by hackerwins — CHANGES_REQUESTED** [review]
Thanks for the contribution. I've commented on some things that need to be fixed.

**Comment by hackerwins** [comment]
It would be nice to consider changing the `timeout-second` variable to duration as well. We can easily convert it using [ParseDuration](https://golang.org/pkg/time/#ParseDuration).

I think it's good to start with a separate issue.

File: pkg/cli/agent.go [hunk 5]
```diff
@@ -120,5 +132,73 @@ func init() {
...
+		&conf.Mongo.ConnectionURI,
+		"mongo-connection-uri",
+		yorkie.DefaultMongoConnectionURI,
+		"MongoDB's connection URI",
+	)
+	cmd.Flags().StringVar(
+		&conf.Mongo.YorkieDatabase,
+		"mongo-yorkie-database",
+		yorkie.DefaultMongoConnectionURI,
+		"MongoDB's connection URI",
```
Comment (dc7303) [inline pkg/cli/agent.go:175]:
```suggestion
		"Yorkie's database name in MongoDB",
```

**Review by dc7303 — CHANGES_REQUESTED** [review]
Thanks for working.
I'd like to ask for a change, so I left a comment. Please confirm.

**Review by hackerwins — APPROVED** [review]
LGTM. Thank you for contributing.

**Pull request merged**